src/main/java/Duke.java

```diff
@@ -1,10 +1,271 @@
+        }
+    }
+
+    public static void updateTask(String[] processedUserInput, int taskNumber, String filePath) throws FileNotFoundException, IOException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        String finalText = "";
+        int lineNumber = 0;
+        while (s.hasNext()) {
+            if (lineNumber == taskNumber) {
+                String oldLine = s.nextLine();
+                String newLine = oldLine.substring(0, 4) + "1" + oldLine.substring(5);
+                finalText = finalText + newLine + "\n";
+            } else {
+                finalText = finalText + s.nextLine() + "\n";
+            }
+            lineNumber++;
+        }
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(finalText);
+        fw.close();
+    }
+
+    public static void removeTask(String[] processedUserInput, int taskNumber, String filePath) throws FileNotFoundException, IOException {
+        File f = new File(filePath); // create a File for the given file path
+        Scanner s = new Scanner(f); // create a Scanner using the File as the source
+        String finalText = "";
+        int lineNumber = 0;
+        while (s.hasNext()) {
+            if (lineNumber == taskNumber) {
+                //Do nothing
+                s.nextLine();
+            } else {
+                finalText = finalText + s.nextLine() + "\n";
+            }
+            lineNumber++;
+        }
+        FileWriter fw = new FileWriter(filePath);
+        fw.write(finalText);
+        fw.close();
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        //Greeting
+        Functions.printGreeting();
+
+        //Setting up task list
+        ArrayList<Task> taskList = new ArrayList<>();
+        int taskListSize = taskList.size();
+        String workingDir = System.getProperty("user.dir");
+        Path filePath2 = Paths.get(workingDir+File.separator+"duke.txt");
+        String filePath = String.valueOf(filePath2);
+        try {
+            taskList = readFileContents(filePath, taskList);
+            taskListSize = getTaskListSize(filePath);
+        } catch (FileNotFoundException e) {
+            //Creates a new duke.txt document
+            File f = new File(filePath);
+            try {
+                f.createNewFile();
+            } catch (IOException ex) {
+                ex.printStackTrace();
+            }
+        }
+
+        //Initialization
+        String userInput = "start";
+        Scanner in = new Scanner(System.in);
+
+        //Main Loop
+        while (!userInput.equals("bye")) {
```

Comment:
Consider converting some of these variables to constants if they are constantly reused. Applies to other parts of your code as well